solarforecastarbiter/datamodel.py

```diff
@@ -1379,17 +1379,19 @@ class QualityFlagFilter(BaseFilter):
         ``DERIVED_MASKS`` keys.
         These periods will be excluded from the analysis.
     """
-    quality_flags: Tuple[str, ...] = (
-        'UNEVEN FREQUENCY', 'LIMITS EXCEEDED', 'CLEARSKY EXCEEDED',
-        'DAYTIME STALE VALUES', 'INCONSISTENT IRRADIANCE COMPONENTS'
-    )
+    type: str
```

Comment:
It isn't ideal, but we should probably keep the `quality_flags` key (and the other parameters would apply to all flags in that tuple). Otherwise all current reports would be broken since I think the report json is loaded through the report datamodel before displaying/downloading.

Reply:
So keep the old `quality_flags` key in addition to the new keys? I agree that could work. A couple of things to consider as alternatives:

1. script to migrate current reports from old json to new json
2. create new objects and json keys, leave the old ones untouched until some later time when we implement 1

I'm not sure what would be less work because the testing implications for keeping the old keys sound miserable.

Reply:
Well keep old key, and add new keys except `type`. So all flags in `quality_flags` are processed with the other options. You would probably have the same options for many of the flags anyway, and you can still have multiple QualityFlagFilters with different sets of flags.

We've done the migration before an it was a real pain (and motivation to eventually get the report schema versioned). One problem is you also have to modify the json core posts to the API. 

I don't think testing would be bad. All the keys are still used (if `type` is removed), we just have a couple of new keys w/ reasonable defaults if empty